base/inference.jl

```diff
@@ -5202,6 +5217,21 @@ function occurs_outside_getfield(e::ANY, sym::ANY,
         if head === :(=)
             return occurs_outside_getfield(e.args[2], sym, sv,
                                            field_count, field_names)
+        elseif head === :foreigncall
+            args = e.args
+            nccallargs = args[5]::Int
+            # Only arguments escape, GC root arguments do not escape.
```

Comment:
ah, the reference to gc made this confusing. but also because the main purpose of the gc-root arguments is to indicate what object(s) need(s) to be pinned in memory during the call. Perhaps a clearer comment might be to mention that this function is computing whether the structure layout is observable from the ccall target, but that the gc-root arguments are not passed through to the child. But also, is that actually true?